File: Algorithms/16. 3Sum Closest.rs
impl Solution {
    pub fn three_sum_closest(nums: Vec<i32>, target: i32) -> i32 {
        let mut sum = nums[0] + nums[1] + nums[2];
        let length = nums.len();
        for i in 0..(length -2) {
            for j in (i + 1)..(length - 1) {
                for k in (j + 1)..length {
                    let temp = nums[i] + nums[j] + nums[k];
                    if (temp - target).abs() - (sum - target).abs() < 0 {
                        sum = temp;
                    }
                }
            }
        }
        sum
    }
}

File: Algorithms/16. 3Sum Closest.ts
function threeSumClosest(nums: number[], target: number): number {
    let result: number = nums[0] + nums[1] + nums[2];
    let nearest: number = Math.abs(target - result);
    nums.sort((a, b) => a - b);
    // console.log(nums);
    for (let i = 0; i < nums.length; i++) {
        let left: number = i + 1;
        let right: number = nums.length - 1;
        while (left < right) {
            const sum: number = nums[i] + nums[left] + nums[right];
            const diff: number = Math.abs(target - sum);
            if (diff < nearest) {
                nearest = diff;
                result = sum;
            }
            if (sum < target) {
                left++;
            } else {
                right--;
            }
        }
    }

    return result;
};
